tests/beacon/test_helpers.py

```diff
@@ -707,6 +722,228 @@ def test_get_domain(pre_fork_version,
+        num_validators,
+        messages[0],
+        privkeys,
+    )
+    valid_params[key] = poc_0_indices
+
+    key = "aggregate_signature_poc_1_indices"
+    # NOTE: does not guarantee non-empty intersection
+    (poc_1_indices, poc_1_signatures) = _get_indices_and_signatures(
+        num_validators,
+        messages[1],
+        privkeys,
+    )
+    valid_params[key] = poc_1_indices
+
+    signatures = poc_0_signatures + poc_1_signatures
+    aggregate_signature = bls.aggregate_signatures(signatures)
+
+    valid_params["aggregate_signature"] = aggregate_signature
+
+    return valid_params
+
+
+def _corrupt_signature(params):
+    params = copy.deepcopy(params)
+    message = bytes.fromhex("deadbeefcafe")
+    privkey = 42
+    domain = SignatureDomain.DOMAIN_ATTESTATION
+    params["aggregate_signature"] = bls.sign(message, privkey, domain)
+    return params
+
+
+def _corrupt_vote_count(params):
+    params = copy.deepcopy(params)
+    key = "aggregate_signature_poc_0_indices"
+    for i in itertools.count():
+        if i not in params[key]:
+            params[key].append(i)
+            break
+    return params
+
+
+def _create_slashable_vote_data_messages(params):
+    # TODO update when we move to `ssz` tree hash
+    votes = SlashableVoteData(**params)
+    return votes.messages
+
+
+def test_verify_slashable_vote_data_signature(privkeys,
+                                              sample_beacon_state_params,
+                                              genesis_validators,
+                                              sample_slashable_vote_data_params):
+    sample_beacon_state_params["validator_registry"] = genesis_validators
+    state = BeaconState(**sample_beacon_state_params)
+
+    # NOTE: we can do this before "correcting" the params as they
+    # touch disjoint subsets of the provided params
+    messages = _create_slashable_vote_data_messages(sample_slashable_vote_data_params)
+
+    valid_params = _correct_slashable_vote_data_params(
+        sample_slashable_vote_data_params,
+        genesis_validators,
+        messages,
+        privkeys,
+    )
+    valid_votes = SlashableVoteData(**valid_params)
+    assert verify_slashable_vote_data_signature(state, valid_votes)
+
+    invalid_params = _corrupt_signature(valid_params)
+    invalid_votes = SlashableVoteData(**invalid_params)
+    assert not verify_slashable_vote_data_signature(state, invalid_votes)
+
+
+def _run_verify_slashable_vote(params, state, max_casper_votes, should_succeed):
+    votes = SlashableVoteData(**params)
+    result = verify_slashable_vote_data(state, votes, max_casper_votes)
+    if should_succeed:
+        assert result
+    else:
+        assert not result
+
+
+@pytest.mark.parametrize(
+    (
+        'param_mapper,'
+        'should_succeed'
+    ),
+    [
+        (lambda params: params, True),
+        (lambda params: _corrupt_vote_count(params), False),
+        (lambda params: _corrupt_signature(params), False),
```

Comment:
this `lambda` isn't needed as I am pretty sure you could just provide `_corrupt_signature` directly.

Reply:
ditto above